ext/pdo_firebird/firebird_statement.c

```diff
@@ -608,6 +608,60 @@ static int firebird_stmt_param_hook(pdo_stmt_t *stmt, struct pdo_bound_param_dat
 				}
 			}
 
+#ifdef SQL_BOOLEAN
+			/* keep native BOOLEAN type */
+			if ((var->sqltype & ~1) == SQL_BOOLEAN) {
+				switch (Z_TYPE_P(parameter)) {
+					case IS_LONG:
+					case IS_DOUBLE:
+					case IS_TRUE:
+					case IS_FALSE:
+						*(FB_BOOLEAN*)var->sqldata = zend_is_true(parameter) ? FB_TRUE : FB_FALSE;
+						break;
+					case IS_STRING:
+						if ((Z_STRLEN_P(parameter) == 0)) {
+							*(FB_BOOLEAN*)var->sqldata = FB_FALSE;
+							break;
+						}
+
+						long lval;
+						double dval;
+
+						switch (is_numeric_string(Z_STRVAL_P(parameter), Z_STRLEN_P(parameter), &lval, &dval, 0)) {
+							case IS_LONG:
+								*(FB_BOOLEAN*)var->sqldata = (lval != 0) ? FB_TRUE : FB_FALSE;
+								break;
+							case IS_DOUBLE:
+								*(FB_BOOLEAN*)var->sqldata = (dval != 0) ? FB_TRUE : FB_FALSE;
+								break;
+							default:
+								if (!zend_binary_strncasecmp(Z_STRVAL_P(parameter), Z_STRLEN_P(parameter), "true", 4, 4)) {
+									*(FB_BOOLEAN*)var->sqldata = FB_TRUE;
+								}
+								else if (!zend_binary_strncasecmp(Z_STRVAL_P(parameter), Z_STRLEN_P(parameter), "false", 5, 5)) {
```

Comment:
Nit: it seems that the rest of this file uses K&R style, i.e. the `else if` should be on the same line as the previous `}`. Same for the `else` a few lines below.